src/test/java/temporary/tests/TestOrderConfirmation.java

```diff
@@ -37,8 +39,8 @@ public void proceedToInvoicePage() {
         updateCart();
```

Comment:
The following code is redundant and can be refactored to an individual method which can be used in the tests to avoid duplication:
openPage();
addToCart();
goToCart();
updateCart();
goToCheckout();
proceedAsGuest();
fillGuestCheckoutForm();
submitGuestCheckoutForm();
confirmOrder();


```

private void completeOrderProcess() {
    openPage();
    addToCart();
    goToCart();
    updateCart();
    goToCheckout();
    proceedAsGuest();
    fillGuestCheckoutForm();
    submitGuestCheckoutForm();
    confirmOrder();
}
```
Use of metod in test

```
 @Test
    @Tag("EndToEnd")
    public void proceedToInvoicePage() {
        // Given:
        completeOrderProcess();

        // When:
        goToInvoicePage();

        // Then:
        validateURLContains("https://automationteststore.com/index.php?rt=account/invoice");
    }
```

Reply:
@TOM-AN1 I could create a test method with all the previous steps of the checkout process and use it in all 3 tests in this class, but I thought that I will continue with the idea from previous tests and put all the steps separately. @Fishgame1 what do you think?

Reply:
You're probably right. Let's stick to the earlier solution and continues it.